Rubrik/Public/Get-RubrikDatabaseRecoveryPoint.ps1

```diff
@@ -0,0 +1,88 @@
+        Format:             Any valid <value> that PS Get-Date supports in: "Get-Date -Date <Value>"
+            Example: "2018-08-01T02:00:00.000Z" restores back to 2AM on August 1, 2018 UTC.
+
+        .NOTES
+        Written by Chris Lumnah for community usage
+        Twitter: @lumnah
+        GitHub: clumnah
+
+        .LINK
+        http://rubrikinc.github.io/rubrik-sdk-for-powershell/reference/Get-RubrikDatabaseRecoverypoint.html
+
+        .EXAMPLE
+        Get-RubrikDatabaseRecoveryPoint -id "MssqlDatabase:::10dd9979-fdcb-4dc2-b212-20efffd39102" -Latest 
+        This will return back a date time combination that prepresents the latest recovery point known to Rubrik about this database. This will include the last snapshot and any transaction log backup taken since that 
+        last snapshot was taken. 
+
+        .EXAMPLE
+        Get-RubrikDatabaseRecoveryPoint -id  "MssqlDatabase:::10dd9979-fdcb-4dc2-b212-20efffd39102" -LastFull
+        This will return back a date time combination that prepresents the recovery point known to Rubrik about this database based on the most recent snapshot taken.
+
+        .EXAMPLE
+        Get-RubrikDatabaseRecoveryPoint -id  "MssqlDatabase:::10dd9979-fdcb-4dc2-b212-20efffd39102" -RestoreTime 02:00:00
+        This will return back a date time combination that represents todays date at 2am, but converted to UTC. The time entered into the RestoreTime field will be in local time. 
+
+        .EXAMPLE
+        Get-RubrikDatabaseRecoveryPoint -id  "MssqlDatabase:::10dd9979-fdcb-4dc2-b212-20efffd39102" -RestoreTime "2019-10-19 20:00:00"
+        This will return back a date time combination that represents the local date time value entered into the RestoreTme field, converted to UTC. 
+    #>
+    param(
+        # Rubrik's database id value
+        [Parameter(
+            Position = 0,
+            Mandatory = $true,
+            ValueFromPipelineByPropertyName = $true)]
+        [ValidateNotNullOrEmpty()] 
+        [string]$id,
+        [Parameter(ParameterSetName = 'Latest')]
+        [switch]$Latest,
+        [Parameter(ParameterSetName = 'LastFull')]
+        [switch]$LastFull,
+        [Parameter(ParameterSetName = 'RestoreTime')]
+        [datetime]$RestoreTime
+    )
+    if ($PSBoundParameters.ContainsKey('Latest')) {
+        $LatestRecoveryPoint = (Get-RubrikDatabase -id $id ).latestRecoveryPoint
+        $RecoveryDateTime = $LatestRecoveryPoint
+    }
+    if ($PSBoundParameters.ContainsKey('LastFull')) {
+        $RubrikSnapshot = Get-RubrikSnapshot -id $id  | Sort-Object date -Descending | Select-object -First 1
+        $RecoveryDateTime = $RubrikSnapshot.date
+    }
+    if ($PSBoundParameters.ContainsKey('RestoreTime')) {
+        $RawRestoreDate = (get-date -Date $RestoreTime)
+        Write-Verbose ("RawRestoreDate is: $RawRestoreDate")
+        $Now = (Get-Date).ToUniversalTime()
+        if ($RawRestoreDate -ge $Now) { 
```

Comment:
@clumnah I added a test to cover the case where the date passed is in the future, but I am curious why this is here in the first place? Is there a reason why a date in the future shouldn't just throw an error?

Reply:
Err - I will merge once @shamsway is ok with this :)